packages/blade-mcp/src/tools/publishLinesOfCodeMetric.ts

```diff
@@ -0,0 +1,119 @@
+import { z } from 'zod';
+import type { ToolCallback } from '@modelcontextprotocol/sdk/server/mcp.js';
+import { analyticsToolCallEventName } from '../utils/tokens.js';
+import { sendAnalytics, handleError } from '../utils/analyticsUtils.js';
+
+// Tool constants
+const publishLinesOfCodeMetricToolName = 'publish_lines_of_code_metric';
+const publishLinesOfCodeMetricToolDescription = `
+Publish the number of lines of code added by the user as a metric.
+
+# When to Use This Tool
+
+Call this tool **immediately after** you (the AI / Cursor) apply a code edit in the consumer’s project.  The caller should send both the number of lines **inserted** (\`linesOfCodeAdded\`) *and* the number of lines **deleted** (\`linesOfCodeRemoved\`) in that single operation.
+
+# Why This Metric Matters
+
+Tracking the net churn (additions vs. removals) per edit helps the Blade team understand adoption, measure productivity boosts, and detect unusually large or small edits that may need review.
+
+# Parameters
+
+• \`files\` — Array of objects, each with:
+  • \`filePath\` — string. Absolute or project-relative path of the file edited.
+  • \`linesAdded\` — integer ≥ 0.
+  • \`linesRemoved\` — integer ≥ 0.
+• \`linesOfCodeAddedTotal\` — Integer ≥ 0.  Sum of \`linesAdded\` across all files.
+• \`linesOfCodeRemovedTotal\` — Integer ≥ 0.  Sum of \`linesRemoved\` across all files.
+• \`currentProjectRootDirectory\` — Absolute path to the consumer’s workspace root (never \`'.'\`).  Used only for analytics aggregation.
```

Comment:
Do we need this parameters section since toolschema basically defines this only.

Reply:
+1